### PR TITLE
add jax.numpy.lexsort and corresponding tests

### LINKED ISSUE #21403: lexsort

### PR DESCRIPTION
**Pull request by illia-bab**
Close #21403

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/backends/paddle/experimental/sorting.py [hunk 1]
```diff
@@ -19,6 +19,13 @@ def invert_permutation(
 def lexsort(
     keys: paddle.Tensor, /, *, axis: int = -1, out: Optional[paddle.Tensor] = None
 ) -> paddle.Tensor:
+    if keys.dtype in [
```
Comment (zaeemansari70) [inline ivy/functional/backends/paddle/experimental/sorting.py:22]: Why are we upcasting these? :slightly_smiling_face: 
Can't we handle this by adding the `unsupported_dtypes` decorators above this function?

Comment (zaeemansari70) [reply ivy/functional/backends/paddle/experimental/sorting.py:26]: According to our discussion as the dtypes suppported by this are `[int32, int64, float32, float64]`. Adding a logic here where we upcast to the closest possible `dtype` would make more sense I guess, what you can do is upcast to `float32` if its a float `dtype` and similarly cast to a closest `int` `dtype` if we get an unsupprted `int` dtype. We will actually be handling a broader set in this way. 
Hope that makes sense :+1:

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Just required a small change

**Comment by zaeemansari70** [comment]
Also, @MahmoudAshraf97 can you take a last look at this as this is related to the paddle backend as also the dtypes handling in paddle backend specifically. 
Thanks!

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
Hello, the new change in the paddle backend doesn't ensure that the returned dtype is the same as the input, please make sure the tests for the functional function is passing in the paddle backend

**Pull request closed without merging**